src/vs/workbench/services/environment/browser/environmentService.ts

```diff
@@ -72,8 +72,8 @@ export class BrowserWorkbenchEnvironmentService implements IEnvironmentService {
 
 		if (remoteUserDataUri) {
 			this.appSettingsHome = remoteUserDataUri || URI.file('/User').with({ scheme: Schemas.userData });
-			this.settingsResource = joinPath(this.appSettingsHome, 'settings.json');
-			this.keybindingsResource = joinPath(this.appSettingsHome, 'keybindings.json');
+			this.settingsResource = joinPath(this.appSettingsHome, 'settings.json').with({ scheme: Schemas.userData });
+			this.keybindingsResource = joinPath(this.appSettingsHome, 'keybindings.json').with({ scheme: Schemas.userData });
 		} else {
 			const appSettingsHome = URI.file('/User').with({ scheme: Schemas.userData });
```

Comment:
@sandy081 what is the purpose of the else branch here when no `remoteUserDataUri` is around? Are you coming up with your own path to settings? Should that be a constant documented somewhere for hosts?

Reply:
This is for the case in web when there is no user data provider and no remote user data uri where I use in memory user data provider.  

Agreed that this is not the right place to set it and my thinking is that environment service is passed in with this information (`appSettingsHome`) while creation. This needs all user data  consumers to be migrated. Hence currently `appSettingsHome` and `settings/keybindings` are having different schemes.